subprojects/core/src/main/java/org/gradle/api/internal/tasks/DefaultTaskContainer.java

```diff
@@ -231,7 +231,7 @@ private <T extends Task> void addTask(T task, boolean replaceExisting) {
             } else {
                 TaskCreatingProvider<? extends Task> taskProvider = Cast.uncheckedCast(findByNameLaterWithoutRules(name));
                 if (taskProvider != null) {
-                    removeInternal(taskProvider);
```

Comment:
Did you see why there is `removeInternal`? Was this to deprecate calling `remove` on `DefaultTaskContainer` directly?

Reply:
You are correct. I was spring-cleaning smaller warnings in this class and inlined the `removeInternal` call because it had a single use and the returned value was not used.

Looking at it now, it better to keep the method to highlight the intent of calling the `super.remove`.